### PR TITLE
GUACAMOLE-337: Move libguacd functionality into public libguac API.

### PR DESCRIPTION
**Pull request by mike-jumper**
This change removes the libguacd utility library, moving its functionality instead into the public libguac library. With these changes, libguac now provides an SSL/TLS implementation of `guac_socket` (if guacamole-server was built with SSL support) and a new `guac_user_handle_connection()` function which deals with the entire Guacamole protocol handshake and subsequent I/O in a blocking manner.

The new SSL/TLS `guac_socket` is defined within the `guacamole/socket-ssl.h` header file (rather than within the main `guacamole/socket.h` header), which will only be installed if SSL support has been included in the guacamole-server build.

guacd has been refactored to take these changes into account.

### CODE REVIEW

**Comment by necouchman** [comment]
I'm good with this, but this is complicated enough that it probably warrants review by someone else.  @jmuehlner??

I'm happy to do the merge work after a review.

File: src/libguac/guacamole/user.h [hunk 1]
```diff
@@ -493,6 +493,30 @@ guac_user* guac_user_alloc();
 void guac_user_free(guac_user* user);
 
 /**
+ * Handles the portion of a user's Guacamole protocol handshake following the
+ * "select" instruction all subsequent I/O, automatically populating the
```
Comment (jmuehlner) [inline src/libguac/guacamole/user.h]: Should there be some more punctuation here or something? This first sentence is really hard to parse.

Comment (mike-jumper) [inline src/libguac/guacamole/user.h]: Nooooo ... my beautiful beautiful sentence ...

I will ... uh ... try to make this make sense.

**Comment by jmuehlner** [comment]
Ok, I think this all looks good now.
